s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -405,3 +456,79 @@ func (t *TidbClient) UpdateUsages(ctx context.Context, usages map[string]int64,
 	}
 	return nil
 }
+
+func buildSql(ctx context.Context, filter map[string]string, sqltxt string) (string, []interface{}, error) {
+	const MaxObjectList = 10000
+	args := make([]interface{}, 0)
+
+	prefix := filter[common.KPrefix]
+	if prefix != "" {
+		sqltxt += " and name like ?"
+		args = append(args, prefix+"%")
+		log.Debug("query prefix:", prefix)
+	}
+	if filter[common.KMarker] != "" {
+		sqltxt += " and name >= ?"
+		args = append(args, filter[common.KMarker])
+		log.Debug("query marker:", filter[common.KMarker])
+	}
+
+	// lifecycle management may need to filter by LastModified
+	if filter[common.KLastModified] != "" {
+		var tmFilter map[string]string
+		err := json.Unmarshal([]byte(filter[common.KLastModified]), &tmFilter)
+		if err != nil {
+			log.Errorf("unmarshal lastmodified value failed:%s\n", err)
+			return sqltxt, args, ErrInternalError
+		}
+		log.Debugf("tmpFilter:%+v\n", tmFilter)
+
+		for k, v := range tmFilter {
+			var op string
+			switch k {
+			case "lt":
+				op = "<"
+			case "gt":
+				op = ">"
+			case "lte":
+				op = "<="
+			case "gte":
+				op = ">="
+			default:
+				log.Infof("unsupport filter action:%s\n", k)
+				return sqltxt, args, ErrInternalError
+			}
+
+			sqltxt += " and lastmodifiedtime " + op + " ?"
+			args = append(args, v)
+		}
+	}
+
+	// lifecycle management may need to filter by StorageTier
+	if filter[common.KStorageTier] != "" {
+		tier, err := strconv.Atoi(filter[common.KStorageTier])
+		if err != nil {
+			log.Errorf("invalid storage tier:%s\n", filter[common.KStorageTier])
+			return sqltxt, args, ErrInternalError
+		}
+
+		sqltxt += " and tier <= ?"
+		args = append(args, tier)
+	}
+
+	delimiter := filter[common.KDelimiter]
+	if delimiter == "" {
+		sqltxt += " order by bucketname,name,version limit ?"
```

Comment:
what is the default order ? what i mean is, how do we ensure the query result from the DB is always deterministic ?

Reply:
Sorry, I missed the order statement, already added now.